crates/bin/node/src/indexer.rs

```diff
@@ -24,16 +28,49 @@ pub async fn init_indexer_task(
 }
 
 pub async fn listen_to_indexer(
+    db_conn: &mut PgConnection,
     mut indexer_service: ApibaraIndexerService,
 ) -> Result<(), ServiceError> {
     info!("Listening indexer events");
+    
     while let Some(event) = indexer_service
         .try_next()
         .await
         .map_err(ServiceError::Indexer)?
     {
+        match event.clone() {
+            Message::Payment(payment_events) => {
+                for payment_event in payment_events {
+                    let amt = sqlx::query_as(
+                        "SELECT amount FROM mint_quote WHERE invoice = ?"
+                    )
+                    .bind(payment_event.invoice_id)
+                    .fetch_all(&db_conn)
+                    .await?
+                    .iter()
+                    .sum();
```

Comment:
do the sum inside the database:

```sql
SELECT SUM(amount) as total_amount
FROM payment_event
WHERE invoice_id = $1
```
call sqlx fetch_one and bind payment_event.invoice_id